Update base version to v0.3.0

mmontour1306 opened the pull request:
This PR ports the Hybrid Compute capabilities onto the v0.3.0 upstream release of Rundler. This was a substantial change from the development snapshot we started with, most notably in that it adds support for the v0.7 Account Abstraction contracts and splits several sections of the code into versioned v0_6/v0_7 branches. The Hybrid Compute logic has only been applied to v0_6 for now.

The upstream release removes the option for a custom EntryPoint contract address, instead hard-coding standard addresses for each contract version. The deploy-local script has been updated to use this address.

I've added https://github.com/safe-global/safe-smart-account as a submodule as it is a dependency for building the v0.6 account-abstraction contracts.

mmontour1306 committed:
Squashed commit of the following:

commit 3ff1330ae7cbba62c96c11f9f3806dce42d4ed36
Author: Michael Montour <mmontour@enya.ai>
Date:   Thu Sep 26 13:39:25 2024 -0700

    Fix unit tests
     Changes to be committed:
    	modified:   crates/rpc/src/eth/api.rs
    	modified:   crates/sim/src/estimation/v0_6.rs
    	modified:   crates/sim/src/estimation/v0_7.rs
    	modified:   crates/types/src/hybrid_compute.rs

commit 2510cde1216498c75934edb8d90957f266225fa4
Author: Michael Montour <mmontour@enya.ai>
Date:   Thu Sep 26 13:08:35 2024 -0700

    Finish merging log msgs etc.
     Changes to be committed:
    	modified:   crates/builder/src/transaction_tracker.rs
    	modified:   crates/provider/src/ethers/provider.rs
    	modified:   crates/rpc/src/eth/api.rs
    	modified:   crates/rpc/src/eth/router.rs
    	modified:   crates/sim/src/simulation/v0_6/context.rs
    	modified:   hybrid-compute/offchain/userop.py

commit 4603b1089a18d31f4527d074f0c6fbcce59df0c1
Author: Michael Montour <mmontour@enya.ai>
Date:   Wed Sep 25 16:28:12 2024 -0700

    More merges.
     Changes to be committed:
    	modified:   ../crates/builder/src/bundle_proposer.rs
    	modified:   ../crates/rpc/src/eth/api.rs
    	modified:   ../crates/rpc/src/eth/events/common.rs
    	modified:   ../crates/sim/src/estimation/estimate_call_gas.rs
    	modified:   ../crates/sim/src/estimation/estimate_verification_gas.rs
    	modified:   ../crates/sim/src/simulation/v0_6/tracer.rs
    	modified:   runit.sh

commit 2dba7c3b170d5e249d4509779d6ad3dddcaea782
Author: Michael Montour <mmontour@enya.ai>
Date:   Tue Sep 24 23:40:04 2024 -0700

    WIP - merge
     Changes to be committed:
    	modified:   .github/workflows/docker-images.yaml
    	modified:   Cargo.lock
    	modified:   Cargo.toml
    	modified:   bin/rundler/chain_specs/boba_sepolia.toml
    	modified:   bin/rundler/chain_specs/optimism_devnet.toml
    	modified:   crates/builder/src/bundle_sender.rs
    	modified:   crates/builder/src/task.rs
    	modified:   crates/provider/src/ethers/entry_point/v0_6.rs
    	modified:   crates/rpc/Cargo.toml
    	modified:   crates/sim/src/gas/gas.rs
    	modified:   crates/types/contracts/src/v0_6/imports.sol
    	modified:   hybrid-compute/offchain/userop.py

commit a488d7c12d926b137f1eedc751dcbc20cc3cabfd
Author: Michael Montour <mmontour@enya.ai>
Date:   Tue Sep 24 22:55:41 2024 -0700

    CORS support.
     Changes to be committed:
    	modified:   crates/rpc/src/task.rs

commit 9963ae009e8e7159411e6884f150480ad50f1679
Author: Michael Montour <mmontour@enya.ai>
Date:   Tue Sep 24 15:43:37 2024 -0700

    Merge .github/.gitignore
     Changes to be committed:
    	modified:   .github/workflows/ci.yaml
    	modified:   .github/workflows/deny.yaml
    	new file:   .github/workflows/docker-images.yaml
    	modified:   .github/workflows/unit.yaml
    	modified:   .gitignore

commit 24c02ee20b9e34346aa35b0429a0c5ae8e3c8978
Author: Michael Montour <mmontour@enya.ai>
Date:   Tue Sep 24 15:30:25 2024 -0700

    Add HC support to bundle_proposer

     Changes to be committed:
    	modified:   crates/builder/src/bundle_proposer.rs
    	modified:   crates/pool/src/mempool/pool.rs
    	modified:   crates/rpc/src/eth/api.rs

commit 112888d5a3446bd8103d0967e724f733cbe8518f
Author: Michael Montour <mmontour@enya.ai>
Date:   Mon Sep 23 15:35:37 2024 -0700

    Update api.rs to new interface.
     Changes to be committed:
    	modified:   crates/provider/src/traits/entry_point.rs
    	modified:   crates/rpc/src/eth/api.rs
    	modified:   crates/rpc/src/eth/router.rs
    	modified:   crates/rpc/src/types/v0_6.rs
    	modified:   crates/sim/src/estimation/mod.rs
    	modified:   crates/sim/src/estimation/v0_6.rs
    	modified:   crates/sim/src/estimation/v0_7.rs

commit d0401797e9c1002970e53d9527a196ccde669005
Author: Michael Montour <mmontour@enya.ai>
Date:   Fri Sep 20 22:30:55 2024 -0700

    WIP checkin
     Changes to be committed:
    	modified:   bin/rundler/src/cli/mod.rs
    	modified:   crates/provider/src/ethers/entry_point/v0_6.rs
    	modified:   crates/provider/src/ethers/entry_point/v0_7.rs
    	modified:   crates/provider/src/traits/entry_point.rs
    	modified:   crates/provider/src/traits/test_utils.rs
    	modified:   crates/rpc/src/eth/api.rs
    	modified:   crates/rpc/src/eth/router.rs
    	modified:   crates/types/src/user_operation/mod.rs
    	modified:   crates/types/src/user_operation/v0_6.rs
    	modified:   crates/types/src/user_operation/v0_7.rs
    	modified:   hybrid-compute/offchain/userop.py

commit a0f2239441ed1a30fc682ddbfa78449131e7302d
Author: Michael Montour <mmontour@enya.ai>
Date:   Thu Sep 19 15:50:42 2024 -0700

    Use standard EntryPoint address. Misc updates.
     Changes to be committed:
    	modified:   bin/rundler/src/cli/chain_spec.rs
    	modified:   crates/types/build.rs
    	modified:   crates/types/contracts/foundry.toml
    	modified:   hybrid-compute/deploy-local.py
    	modified:   hybrid-compute/local.env

commit acdccbf823ba6e79d5746a85c30cdfdd94861425
Author: Michael Montour <mmontour@enya.ai>
Date:   Thu Sep 19 13:15:04 2024 -0700

    Add chain IDs for Boba Sepolia and local devnet.
     Changes to be committed:
    	new file:   bin/rundler/chain_specs/boba_sepolia.toml
    	new file:   bin/rundler/chain_specs/optimism_devnet.toml

commit 4376ef30d19a81b7a48060eebdd6a89fceff77b7
Author: Michael Montour <mmontour@enya.ai>
Date:   Thu Sep 19 12:52:15 2024 -0700

    Update deployer to work with new contract paths.
     Changes to be committed:
    	new file:   crates/types/contracts/hc_scripts/DeployHybridAccount.sol
    	new file:   crates/types/contracts/hc_scripts/ExampleDeploy.s.sol
    	new file:   crates/types/contracts/hc_scripts/LocalDeploy.s.sol
    	modified:   hybrid-compute/deploy-local.py

commit 503da92bc146bbda469c87f5731b2c6971e645b9
Author: Michael Montour <mmontour@enya.ai>
Date:   Tue Sep 17 22:46:21 2024 -0700

    Update submodule.
     Changes to be committed:
    	modified:   .gitmodules

commit 48e3e732036434fad2ca24747494cefa4b2ba8ee
Author: Michael Montour <mmontour@enya.ai>
Date:   Tue Sep 17 22:40:32 2024 -0700

    Change the runit.sh wrapper
     Changes to be committed:
    	modified:   hybrid-compute/runit.sh

commit 3bd1db78bb19ecd4c61bcf89336dd026f54a92d6
Author: Michael Montour <mmontour@enya.ai>
Date:   Tue Sep 17 22:40:00 2024 -0700

    Add the hybrid_compute.rs file, updated for new UserOperations.
     Changes to be committed:
    	modified:   Cargo.lock
    	modified:   crates/types/Cargo.toml
    	new file:   crates/types/src/hybrid_compute.rs
    	modified:   crates/types/src/lib.rs

commit 867d9fe74dfc05ea29ab622dec3f58beff3e0517
Author: Michael Montour <mmontour@enya.ai>
Date:   Tue Sep 17 14:52:01 2024 -0700

    Import the hybrid-compute directory.

     Changes to be committed:
    	new file:   hybrid-compute/Dockerfile.offchain-rpc
    	new file:   hybrid-compute/README.md
    	new file:   hybrid-compute/aa-client.py
    	new file:   hybrid-compute/aa_utils/__init__.py
    	new file:   hybrid-compute/deploy-local.py
    	new file:   hybrid-compute/docker-compose.yml
    	new file:   hybrid-compute/local.env
    	new file:   hybrid-compute/offchain/add_sub_2/add_sub_2_offchain.py
    	new file:   hybrid-compute/offchain/add_sub_2/add_sub_2_test.py
    	new file:   hybrid-compute/offchain/add_sub_2/readme.md
    	new file:   hybrid-compute/offchain/auction_system/auction_system_offchain.py
    	new file:   hybrid-compute/offchain/auction_system/auction_system_test.py
    	new file:   hybrid-compute/offchain/check_kyc/check_kyc_offchain.py
    	new file:   hybrid-compute/offchain/check_kyc/check_kyc_test.py
    	new file:   hybrid-compute/offchain/check_kyc/readme.md
    	new file:   hybrid-compute/offchain/get_token_price/get_token_price_offchain.py
    	new file:   hybrid-compute/offchain/get_token_price/get_token_price_test.py
    	new file:   hybrid-compute/offchain/get_token_price/readme.md
    	new file:   hybrid-compute/offchain/offchain.py
    	new file:   hybrid-compute/offchain/offchain_utils.py
    	new file:   hybrid-compute/offchain/rainfall_insurance/rainfall_insurance_offchain.py
    	new file:   hybrid-compute/offchain/rainfall_insurance/rainfall_insurance_test.py
    	new file:   hybrid-compute/offchain/rainfall_insurance/readme.md
    	new file:   hybrid-compute/offchain/ramble/ramble_offchain.py
    	new file:   hybrid-compute/offchain/ramble/ramble_test.py
    	new file:   hybrid-compute/offchain/ramble/readme.md
    	new file:   hybrid-compute/offchain/readme.md
    	new file:   hybrid-compute/offchain/sports_betting/sports_betting_offchain.py
    	new file:   hybrid-compute/offchain/sports_betting/sports_betting_test.py
    	new file:   hybrid-compute/offchain/userop.py
    	new file:   hybrid-compute/offchain/userop_utils.py
    	new file:   hybrid-compute/offchain/verify_captcha/__init__.py
    	new file:   hybrid-compute/offchain/verify_captcha/captcha_offchain.py
    	new file:   hybrid-compute/offchain/verify_captcha/captcha_test.py
    	new file:   hybrid-compute/offchain/verify_captcha/readme.md
    	new file:   hybrid-compute/runit.sh

commit a88a4c77160d098dc1ea7ec665e9998c9304fb4c
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Mon Jul 1 16:36:00 2024 -0500

    feat: add support for avax chains

    - Add gas overheads to chain spec (#638)
    - Allow for retries in chain sync
    - Add avax and avas fuji chain specs

commit 459dab6fab992433f931b9f0779b4835eb6695d2
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jul 2 14:19:45 2024 -0500

    feat(sim): ban access to arbitrum stylus contracts during sim

commit 111868391c95c05e74edece693964161dc282ed7
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jul 2 16:50:09 2024 -0500

    fix(builder): improve replacement underpriced handling

commit f7aa69448f36154537a7c34647759ff191d5afdb
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jul 2 11:31:58 2024 -0500

    fix(cli): error if recursive base config, remove goerli netowrks

commit 307e8c2cbeab80518a56a2f309e994de3b1631f0
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Fri Jun 28 14:56:29 2024 -0400

    feat(tx): change transaction size limit (#749)

commit 214e550b95fa7e82b83033a62959466a244c871c
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Jun 28 10:53:40 2024 -0500

    fix: various v0.3-rc fixes

commit b97292a6004460eeb3eb724f241c41edbe01de9b
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jun 25 14:05:41 2024 -0500

    chore: update to rust v1.79.0

commit 65ee1e0d02d64246500fd962bb358f917b29b50c
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Jun 27 08:02:14 2024 -0500

    chore(amoy): raise amoy chain config min fee to 30gwei

commit 9a84f6e2b2bec12d802c2f752bf34e7273dbc8f1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Wed Jun 26 16:16:51 2024 -0500

    fix(pool): fix crash when block height goes backwards

commit 3f364569c30ee1a57ab628370be17681125e2307
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jun 25 14:05:41 2024 -0500

    chore: fix docs and set v0.3 version number

commit 7611b13e1495cffd83414059d130fd763a87d6e1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jun 25 13:20:38 2024 -0500

    fix(rpc): fix error message for unstaked entity mempool count

commit 2706f46fa0295c1cc377f9c0bccadbc4ad949c65
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Mon Jun 24 12:47:19 2024 -0500

    fix(builder): fix builder state machine replacement tracking

commit 85f1897c117712b2a9a77272a1b95ec9f1142b6e
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Mon Jun 24 14:54:28 2024 -0500

    feat(sim): check total gas limit after estimation

commit 1dca183002b89c12f61750018ed77719451fcffb
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jun 18 22:03:00 2024 -0500

    chore(builder): add unit tests to bundle_sender

commit f3a9c9849fe0cb69040eec07e14195518196742b
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Jun 14 10:40:34 2024 -0500

    feat(builder): allow for multiple private keys

commit f8c59e75450f6d3ed33a1d93781599f7e1b3e11f
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Jun 13 17:15:09 2024 -0500

    feat(builder): reject ops if condition not met after failure

commit 91ccf11f6e1f2e36c9d79b0cc383855f38f93240
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Wed Jun 12 16:55:55 2024 -0500

    fix(builder): rework raw sender to support dropped/conditional/split-rpcs correctly

commit 064d38c80b1c72aaaf70c216c3e95cb0b020bc88
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jun 11 14:59:56 2024 -0500

    refactor(builder): large refactor of bundle sender state machine and metrics

commit 11c47b280d23fbe9c887be93376a4ccf72199a31
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jun 11 11:56:55 2024 -0500

    feat(builder): cancel transaction when replacement underpriced

commit a72cdf3aeb214adcbd0f1bc01705c105a3f5fa6a
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Wed Jun 5 15:41:38 2024 -0500

    feat(builder): remove polling from transaction tracker, update state machine

commit 39edc28c16ae7de57ac114d59f28164628a9d7a3
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Jun 14 12:23:53 2024 -0500

    feat(pool): allow staked senders to have multiple UOs in best_operations

commit e41f1e55954dfdd6082b837b29dab892101a1785
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Jun 18 16:02:51 2024 -0500

    feat(pool): add time to mine tracking and metrics

commit 80129e637c843c83d2b30370688fd50e617837be
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Jun 14 16:18:06 2024 -0500

    fix(pool): fix race condition in paymaster tracking

commit 416f2636ebe123d587526d3ba2a4c221d27cb5af
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Tue Jun 18 18:49:04 2024 -0400

    fix(tracer): set tracer timeout default back to 10s (#733)

commit 0d1683f6a839989010f567977025c4527feb6db9
Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Date:   Tue Jun 18 20:17:59 2024 +0000

    chore(deps): bump braces from 3.0.2 to 3.0.3 in /crates/sim/tracer

    Bumps [braces](https://github.com/micromatch/braces) from 3.0.2 to 3.0.3.
    - [Changelog](https://github.com/micromatch/braces/blob/master/CHANGELOG.md)
    - [Commits](https://github.com/micromatch/braces/compare/3.0.2...3.0.3)

    ---
    updated-dependencies:
    - dependency-name: braces
      dependency-type: indirect
    ...

    Signed-off-by: dependabot[bot] <support@github.com>

commit 23da1225ed87bb2677abc7a575cfa75c54f280f7
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Tue Jun 18 16:17:14 2024 -0400

    feat(tracer): add timeout of 10s to tracer (#730)

commit 98f8e2ccd1da6513822e6f76097c3a6624cc6129
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Fri Jun 14 14:38:20 2024 -0400

    feat(ci): check if output file exists (#727)

commit 6c96ed6fd7179df2cad0f526fc8ffd46a73b886b
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Jun 7 16:42:48 2024 -0500

    chore: release v0.2.2

commit 1e8f5e206a73600935fcee062c7390eacf665251
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Jun 7 16:19:36 2024 -0500

    fix(cli): fix ep disable flags

commit 0074fe020b1946f30d0e71da89776d691a44ba24
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Jun 7 09:19:00 2024 -0500

    chore: release 0.2.1, small doc fix

commit 7990f85f4dd0f6ac22a9a76240f2eebaf023cc8e
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Mon Jun 3 21:30:00 2024 -0500

    fix(sim): handle v0.7 executeUserOp special call in call gas estimation

commit 07f66997e164bf7bff7f43929404f1ab33811b2d
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Mon Jun 3 14:06:10 2024 -0500

    fix(cli): add missing global to many common parameters

commit 3b4a4aa444762199fa245c6a0193ab3a41bc362c
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri May 24 09:38:59 2024 -0500

    chore: release v0.2.0

commit 78f15c962576cdb23948b658859ba696e1d88a8e
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Fri May 24 11:06:53 2024 -0400

    feat(simulation): check whether the op fits within a valid time range (#712)

commit 332d24102c0adc4c39cf59b820653fead05c8096
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu May 23 07:58:59 2024 -0500

    fix(sim): update simulation for dencun opcodes

commit 02af86989ec90045bbee7e2890062128665adf3b
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu May 16 13:51:07 2024 -0400

    fix(builder): filter fee before filtering by gas limit

commit 7cc3f5f3a7430d36184f17cbf1c989de09cb8d95
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri May 10 14:40:13 2024 -0500

    chore(docs): fix dead ERC-4337 links

commit c1dc6a91015deea5fa018c08956ecfa673902da1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri May 10 14:14:48 2024 -0500

    fix(provider): use state overrides for gas used, not ctor method

commit 58f31f425dade4af4e59cdfc0da9e42ea795999e
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu May 2 15:43:30 2024 -0500

    fix(chain): set maximum transaction size bytes to 128KB for ethereum chains

commit 574038aa7313b322c62d3089cf524c8520ae66aa
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu May 2 12:16:25 2024 -0500

    chore(docs): add docs for entry point version support

commit 5cdcc4686e428909ce788eae7fc055fecfe1e988
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu May 2 09:17:41 2024 -0500

    fix(types): define entrypoint specific gas overheads

commit d1a33937433587c232727f1facc4c66200690a66
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Apr 30 12:36:34 2024 -0500

    fix(types): fix unpack factory from initcode on v0.7

commit edab84ce8762eeec65a69142dd729bc77acadeb1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Apr 26 11:28:30 2024 -0500

    feat(builder): support multiple flashbots builders

commit fb32d1705bc9d750bd01301714b6c14eef556a1a
Author: Alex Miao <alex.miao@alchemy.com>
Date:   Tue Feb 20 11:05:51 2024 -0800

    fix: add flashbots signature header

commit a32b6f75f9a9f327232dec63a71121839bdb02e0
Author: Alex Miao <alex.miao@alchemy.com>
Date:   Wed Feb 14 16:57:23 2024 -0800

    refactor: use flashbots relay rpc

commit 50d4b1679431d081e902edf69de16fb51b244764
Author: Alex Miao <alex.miao@alchemy.com>
Date:   Thu Feb 1 18:37:17 2024 -0800

    feat(builder): add option to pass in additional builders to flashbots mev-share rpc

commit c0d5235b2b8d4e7a407702ad5661622e41584a38
Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Date:   Wed Apr 24 18:29:04 2024 +0000

    chore(deps): bump rustls from 0.22.2 to 0.22.4

    Bumps [rustls](https://github.com/rustls/rustls) from 0.22.2 to 0.22.4.
    - [Release notes](https://github.com/rustls/rustls/releases)
    - [Changelog](https://github.com/rustls/rustls/blob/main/CHANGELOG.md)
    - [Commits](https://github.com/rustls/rustls/compare/v/0.22.2...v/0.22.4)

    ---
    updated-dependencies:
    - dependency-name: rustls
      dependency-type: direct:production
    ...

    Signed-off-by: dependabot[bot] <support@github.com>

commit 2cba170c94df87c4499e9bf5465adf2ce9235045
Author: 0xfourzerofour <joshpauline@gmail.com>
Date:   Wed Apr 24 11:44:15 2024 -0400

    feat(entrypoint): use checksummed address with correct casing

commit 2231bacda7842dc728d5f6df45ea9a43d40a1f65
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Wed Apr 24 09:33:31 2024 -0500

    fix(sim): handle explicit zero fees in 0.7 pre-verification gas

commit a42c74839de321e1b44ba4821425b5a29f213de5
Author: David Philipson <david.philipson@gmail.com>
Date:   Tue Apr 23 22:05:11 2024 -0700

    fix(sim): handle explicit zero fees in 0.6 pre-verification gas

    Treat explicitly passed zero values the same as absent values for gas
    fees in `preVerificationGas` estimation in v0.6, as is consistent with
    estimating gas limits.

commit 9ba06103b7fe7c2aed5a80fee5e959b6e7117d65
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Wed Apr 24 09:30:37 2024 -0500

    chore(sim): move bytecode to contracts crate, use hex constant

commit 75180322e7a6dd90a20fc942549220cf48d78c3a
Author: David Philipson <david.philipson@gmail.com>
Date:   Tue Apr 23 16:31:28 2024 -0700

    fix(simulation): don't use abi-generated EntryPoint bytecode

    This reverts commit 275414d9268997fa9519cdf77d56f88a202e523d.

    We need the v0.6 entry point bytecode so we can use it in state
    overrides during call gas estimation. In a previous commit, we started
    using bytecode generated by abigen to avoid hardcoding the entry point
    bytecode or fetching it from on-chain on every run. For unknown reasons,
    the abigen bytecode is faulty and causes reverts with no data, so switch
    back to using hardcoded bytecode.

commit 8eb4cb4c93ffb513a45522ceaa999fda108fff6f
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Apr 23 10:12:52 2024 -0500

    fix(cli): enforce num keys greater than or equal to num builders

commit f2f3dd87f00e426943db4a6a4bb272236acc0ff5
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 18 20:32:47 2024 -0500

    fix(rpc): make the validationRevert error message more clear

commit 4ef403c0ebfc15284a12e073b0c2de9ec5fc0dce
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 18 15:23:08 2024 -0500

    feat(rpc): return internal errors and increment metric on panic

commit 49a9ccb59bb2dc7c92e82ed8a8c891bcb30c09e9
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 18 08:51:39 2024 -0500

    feat(rpc): add distinct types for EP version gas estimations

commit b266807479390e5045672fb5a9849dec6d3b4034
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Apr 16 20:10:05 2024 -0500

    feat(sim): skip estimation if gas field is provided in op

commit 71242888ff2472c65cb90087c4f71d523d51faae
Author: David Philipson <david.philipson@gmail.com>
Date:   Tue Apr 16 15:07:07 2024 -0700

    chore(docs): Update commands in makefile, readme, and help messages

commit 26a8fa2f681dd1026e4c0c0267ffdc1197b19fa9
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Apr 16 18:44:45 2024 -0500

    fix(rpc): include inner error message in revert message

commit 32557bd0124e1310a73059cc5079fbb5f183ea3c
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Apr 16 17:35:42 2024 -0500

    fix: add abi_encoded_size back to UserOperation

commit e70538a1eb20f522ac3628e98f39fc01bd920b4a
Author: David Philipson <david.philipson@gmail.com>
Date:   Mon Apr 15 17:27:57 2024 -0700

    refactor(simulation): use abi-generated EntryPoint bytecode

    Rather than hardcoding v0.6 entry point bytecode in two places, one with
    the init code and one with the deployed code, use abigen to generate
    Rust code for the v0.6 `EntryPoint.sol` and use the bytecode that was
    generated.

    This makes v0.6 `IEntryPoint.sol` unnecessary, to remove it and convert
    all occurrences of `v0_6::i_entry_point::IEntryPoint` to
    `v0_6::entry_point::EntryPoint`.

commit 3b6bf73a8490078f33d133f1e3df2ee2eb315cf1
Author: David Philipson <david.philipson@gmail.com>
Date:   Wed Apr 3 00:19:11 2024 -0700

    feat(simulation): v0.7 call gas estimation

    Factors out the call gas estimation logic, which performs a binary
    search using the `target` and `targetData` arguments to
    `simulateHandleOp` along with some state override trickery, into a new
    trait, `CallGasEstimator`, then provides implementations for v0.6 and
    v0.7.

commit 7f4804ecc941ad5447fd8223f7a0194b2d928e51
Author: David Philipson <david.philipson@gmail.com>
Date:   Fri Apr 12 00:11:34 2024 -0700

    feat(rpc): return validation revert data from rpc

    When validation fails in 0.7 in either simulation or gas estimation, the
    entry point now surfaces the inner revert which caused validation to
    fail. Surface this in the error response in the rpc, using the `data`
    field to return an object with the fields:

    - `entry_point_reason`: the entry point string like "AA99 some failure".
      This is the only field populated in v0.6.
    - `inner_revert_bytes`: the raw bytes of the inner revert message.
    - `inner_revert_reason`: if the inner revert is caused by a Solidity
      `require` or `assert`, then the string message provided to it.

commit 9e88be2599e1204c426dd05f9266afd64a504c41
Author: David Philipson <david.philipson@gmail.com>
Date:   Fri Apr 12 00:37:27 2024 -0700

    chore(simulation): remove postop gas estimates

commit b10a43926839941363bc6d3ceb01a1afecbf80b0
Author: David Philipson <david.philipson@gmail.com>
Date:   Sun Mar 31 19:21:54 2024 -0700

    feat(simulation): v0.7 verification gas estimation

    Factors out the verification gas estimation logic, which performs a
    binary search by making many `eth_call`s in a row, into a new trait,
    `VerificationGasEstimator`, then provide implementations for v0.6 and
    v0.7.

    Some notable fixes along the way:

    * Change the Rundler settings options around gas limits. There are now
      separate gas limits for all four things requiring gas in v0.7.
    * Fixed some logic around v0.7's `simulateHandleOp`, which, unlike its
      counterpart in v0.6, returns its results normally rather than by
      reverting.
    * Add some more functionality in and around `UserOperationBuilder` to
      facilitate updating fields in an existing `UserOperation`.

commit e311a8cf34ed9718e012a47fd1925ac96d6c9d1a
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Apr 12 06:37:34 2024 -0500

    chore(ci): fix remote tests for v0.6 and v0.7

commit 14b818ed383ebbed8137d6ee4623018350a2ec72
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Apr 12 11:12:53 2024 -0500

    fix(sim): use same method for expected storage in v0.7 from v0.6 tracer

commit 2e4188637d3eda68296b16f40a17b723b2093518
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 11 20:33:23 2024 -0500

    feat(sim): sto-022 factory or entity needs stake, with cleanups

commit 50d632667750ea4bf267142eefe057646a0f6d90
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 11 18:56:35 2024 -0500

    feat: add entry point to some logs and metrics

commit 0de2be4913a8837affd92dc33d90156cc6a6fac0
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Wed Apr 10 19:32:46 2024 -0500

    fix(sim): use H256 instead of U256 for expected storage

commit 2b82bb5045cb76bab4f3f4dee1459af99c76219e
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 4 16:13:57 2024 -0400

    feat(sim): correctly attribute OP-054 and OP-061 to the correct phase

commit 531b6b6e74e7a43ebfb92b40007c3af3ce4ca957
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 4 14:35:24 2024 -0400

    feat(sim): cleanups for v0.7 sim

commit 6e91e82d6dd174d70bc3af26cc87bbe8a10ff1e1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Wed Apr 3 14:22:04 2024 -0500

    feat(sim): implement simulation for v0.7 entrypoint using reference tracer

commit f99c68c2582e1c06e3abe115d6119eb1085f3942
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Apr 2 09:04:15 2024 -0500

    refactor: create context interface for simulation

commit 97d257558ad7500d07bc983d9b5c923e23991a10
Author: David Philipson <david.philipson@gmail.com>
Date:   Mon Apr 1 17:57:59 2024 -0700

    fix(mempool): fix configuration deserialization

    Correctly deserialize the `entryPoint` field, and display a more
    descriptive error when deserialization fails.

commit 0b65b981dfc61093d13165a6718321b0e9a5daed
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Sun Mar 31 20:31:23 2024 -0500

    feat: cleanups for entry point v0.7 support

commit b7753661624ec03850496858dab138a947238678
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Sun Mar 31 09:36:25 2024 -0500

    chore: fixing PR comments

commit 89f527bcb4daf70a4a954b4889f624be6a2f8d55
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Sat Mar 30 13:26:17 2024 -0500

    chore: fixing PR comments

commit 3e2b0eb356a58d143bedaef0267dfaf7796f8c62
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Sat Mar 30 13:18:03 2024 -0500

    fix: remove extra box from dyn objects

commit 03790598cb3c61da4d792f86e97df1a94d57bb42
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Sat Mar 30 12:07:28 2024 -0500

    chore: fixing PR comments

commit 0983c536141a9a7357e11ffec8381a7fdd8fbc9c
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Fri Mar 29 16:55:23 2024 -0500

    feat: end to end entry point routing (#649)

commit d49e9ade706456ac463c3007f6964d20627f1bbe
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Fri Mar 29 16:48:46 2024 -0500

    feat: unsafe mode for EP v0.6 (#648)

commit 2353c281950190a953f379ad53a5f27073b3ccf1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Mon Mar 25 16:20:10 2024 -0400

    feat(provider): add an entry point v0.7 provider

commit b3b16df65daf84070476b6338e25d6478a503f6d
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Fri Mar 29 16:49:26 2024 -0500

    feat(builder): start support for multiple entry points in builder (#645)

commit 25d9bdf8ba1c95aedb8736fe92b8041406779856
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Mar 21 20:21:36 2024 -0400

    feat(pool): Add entry point routing to pool

commit ff93640c5083d6b9c4039cafd785167f8e2e7a3c
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Mar 14 18:48:14 2024 -0400

    refactor: move builder and pool traits to types

commit 7496d732dcd06def7f55e58df24fd041206e51a1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Mar 14 13:53:42 2024 -0400

    feat(rpc): introduce entry point routing

commit 552ee6c9e94245af48c0d0d9aac42a2267f93868
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Mar 12 06:59:12 2024 -0400

    feat: add and use a user operation trait

commit 532739f4417ab636eb271803e9708e5e566c31c1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Mar 5 17:01:43 2024 -0500

    feat(types): add entrypoint 0.7 contract types

commit de9e409edff3edf0bec6aa685363356b1bd32ff1
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Mar 12 06:59:12 2024 -0400

    feat: add and use a user operation trait

commit 9ecced13a1713a3e36ff1e04e09e38666ad622a0
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Tue Mar 5 17:01:43 2024 -0500

    feat(types): add entrypoint 0.7 contract types

commit 979f15b7ba7fe6ad250c11c1539fa6027b9e178d
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Apr 19 12:29:34 2024 -0500

    chore: start v0.2.0-rc0

commit b3c71ffea3713fb917ccd65bb4d46b0eb8fd5dc5
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Apr 19 12:12:05 2024 -0500

    chore: fix v0.1.0 version

commit b12f683d5565a49a56feea0756058bc82ef52c4d
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Fri Apr 19 07:37:00 2024 -0500

    fix(builder): add builder index to tracker metrics

commit 29409d75241d553863196758cf26a7331d555f10
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 18 11:17:39 2024 -0500

    fix(bin): update metrics-process version to fix metrics

commit cfff5a378ccc63cc77067320f83d652bb610ca19
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 18 09:25:40 2024 -0500

    fix(builder): add missing builder index to metrics

commit 68e6eaba172f2a274137dda0262290105740c5b7
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Tue Apr 16 15:37:24 2024 -0400

    feat(proposer): add field in chainspec to limit size of bundle in bytes  (#681)

commit 45f39d8e28a89466488b10384359351e8bc2ab15
Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Date:   Fri Apr 5 16:15:51 2024 +0000

    chore(deps): bump h2 from 0.3.24 to 0.3.26

    Bumps [h2](https://github.com/hyperium/h2) from 0.3.24 to 0.3.26.
    - [Release notes](https://github.com/hyperium/h2/releases)
    - [Changelog](https://github.com/hyperium/h2/blob/v0.3.26/CHANGELOG.md)
    - [Commits](https://github.com/hyperium/h2/compare/v0.3.24...v0.3.26)

    ---
    updated-dependencies:
    - dependency-name: h2
      dependency-type: indirect
    ...

    Signed-off-by: dependabot[bot] <support@github.com>

commit 4042160b046698b8032f8e98f6f0e1875fee2124
Author: clonefetch <c0217@outlook.com>
Date:   Tue Apr 9 10:48:32 2024 +0800

    chore: remove repetitive words

    Signed-off-by: clonefetch <c0217@outlook.com>

commit fc79aa49a3b7cd1bf5ecde296aad4d2664181a74
Author: dancoombs <dan.coombs@alchemy.com>
Date:   Thu Apr 11 16:47:06 2024 -0500

    fix(estimation): handle zero fee fields in pvg estimation

commit 0054d2b5029515e2f1a2a252bf716c8a0d2f6752
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Mar 28 13:32:49 2024 -0500

    fix(sim): fix sto-022 (#655)

commit dc8f6e2a912f59c0941742389310b81c0d171f38
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Mar 28 07:24:05 2024 -0500

    feat(rpc): use zero fees in no paymaster estimation case (#651)

commit 4e69bd3421f1d190220759e5fd34771456013d9c
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Wed Mar 27 18:42:37 2024 -0400

    fix(sim): update storage access rules to return correct entity (#647)

commit 3d9ac77eaf36ea1dcd90283685e6cc7a4ec9e257
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Wed Mar 27 15:55:20 2024 -0400

    feat(spec): add compliance back (#631)

commit adebb49be112adb34b3d5562793397706bccc2e8
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Mon Mar 25 13:51:54 2024 -0400

    fix(sim): account for native transfer overhead in estimation (#643)

commit e7cbc3c3a3ac89c8a221cb2fb7843c47f5de1993
Author: Leo Antelyes <lantelyes@gmail.com>
Date:   Mon Mar 25 22:56:15 2024 +0900

    fix(docs): add missing command to full node docker-compose.yaml example (#632)

commit 7dd8f3676bae1e610b3d494cfa48916f2990517b
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Mar 20 19:42:10 2024 -0400

    fix: fix the error code for execution failed in RPC metrics (#642)

commit 8bb5886f4ea41271885dde636539c626ac97d863
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Mar 20 16:15:30 2024 -0400

    fix(sim): handle aa51 as OOG in estimation (#641)

commit 13fd5fd4621c224c905e3966a990dd984351bdbd
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Mar 13 12:33:47 2024 -0400

    feat(sim): add support for op-080 (#637)

commit 69ccf3c80f7386a298d467d97862c7eb38a9d347
Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Date:   Mon Mar 4 18:38:36 2024 -0500

    chore(deps): bump mio from 0.8.10 to 0.8.11 (#627)

    Signed-off-by: dependabot[bot] <support@github.com>
    Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

commit 725b11562eef64da3359b5a8807327e3ed9c4a3d
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Mon Mar 4 18:18:16 2024 -0500

    chore(ci): fix remote spec tests (#629)

commit 555d230b0ae010990bb86c1331ffd2d1d780651a
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Mon Mar 4 17:45:45 2024 -0500

    fix(rpc): add a buffer to call gas limit estimation (#628)

commit 8702543af007d6504bf7c41bd20208bdf5bd952e
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Mon Mar 4 17:30:26 2024 -0500

    feat(release): update version (#622)

commit fc82b4d368b43f6e62bb95feb0b77bf030f01e5b
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Fri Feb 23 11:18:22 2024 -0500

    chore: change version to 0.2.0-alpha.0 (#621)

commit 899794ecada8890267e4b2bfb7696736e74815ad
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Fri Feb 23 11:07:29 2024 -0500

    feat(cross-arch): add cross arch release pipeline (#487)

commit a5cde2d80a2c61a28db2ac011c1e73d9f071d52b
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Thu Feb 22 21:46:03 2024 -0500

    feat(metrics): add internal rpc metrics  (#618)

commit 475b47d05c77e288ddd58051df06f8d73a73e511
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Feb 22 19:07:08 2024 -0500

    chore(docs): add docs for debug_bundler_getStakeStatus (#616)

commit ded162c8685a842c29817953379f00716fd391b8
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Feb 22 19:02:54 2024 -0500

    chore: update deps (#615)

commit 1b6cf2436842960591fa67ec03ca8341b1cd84d0
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Feb 22 18:52:31 2024 -0500

    feat(pool): add paymaster lru cache (#614)

commit da6cc016e20ee636fcdac828f601d8b805c1893d
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Feb 22 18:47:43 2024 -0500

    refactor: modify how balances are batch read (#613)

commit ffec1410457580b77a17cf273cc7cf4f24ab52e7
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Feb 22 16:34:29 2024 -0500

    fix(pool): do chain updates async (#619)

commit 44fae8a9c8d1910f200329b00bc6cc65262a6f56
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Feb 21 15:08:34 2024 -0500

    feat: add drop local user operation endpoint (#610)

commit 20b725fea994490a1c3d716c3c51137b8e57ff54
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Wed Feb 21 11:37:55 2024 -0500

    feat(paymaster): refactor paymaster and reputation structure (#577)

commit 21797d59513d532928ef2b221b8fd1539d87639b
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Feb 21 10:32:27 2024 -0500

    feat: add debug rpc method to dump paymaster balances from pool (#609)

commit 06b76f2dba5a6e51d10b89e91f8b4c7497842d58
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Tue Feb 20 19:36:09 2024 -0500

    chore(bin): remove base goerli chain spec (#612)

commit abe73edd567782c4f0f0a00a88fe9eb7bf06255a
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Tue Feb 20 19:16:10 2024 -0500

    fix(builder): move timer interval creation outside of loop (#611)

commit 402528fde1cb76e1ec0683867b7229b3c8a88d1e
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Tue Feb 20 15:46:20 2024 -0500

    feat(builder): add a timer based trigger to bundle building (#607)

commit 88bdb7317d54b2ade99700af32fbe95e66ef7f9d
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Fri Feb 16 17:11:29 2024 -0800

    chore(docs): add chain spec docs (#606)

commit 4cb00d6ce3d6c6e2838118f718b46b2a6ba334eb
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Feb 15 15:17:46 2024 -0800

    fix(rpc): enforce min in priority fee estimation (#605)

commit 8c251fd18f6f4171ad80597ea4add5e394c51cbb
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Feb 15 11:36:02 2024 -0800

    feat: add chain spec, use it for gas, add configs (#549)

commit 7e51896db9bd48c43afa7171cba738b65cb196ad
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Thu Feb 15 08:34:30 2024 -0800

    feat(sim): reject UOs that don't have a buffer on verification gas limit (#599)

commit 30674eef8f191ecb3dd9e4369a19b7cf713f8154
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Feb 14 18:40:55 2024 -0800

    chore: update spec tests after rep fix (#602)

commit 69854a73df8e2b22d1e506e2a4889de94cc35321
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Feb 14 18:38:04 2024 -0800

    fix(pool): fix remote pool error conversion (#601)

commit ff6889c52fbde927853cab9787c4ca3dd014a9d7
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Feb 14 17:52:06 2024 -0800

    fix(pool): fix proto reputation status conversion (#600)

commit bb6e28adaf4bda904d28cbc298cce0250cd9ab76
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Wed Feb 14 10:51:03 2024 -0800

    chore(spec): update spec tests to v0.6 (#594)

commit e73e2e5a373bd7282d4708f22d42997d9468e2e6
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Tue Feb 13 15:52:42 2024 -0800

    fix(rpc): handle double postop revert data parse (#592)

commit cda0c81c8457ae495d619ca29f8d2dc04dc3a242
Author: Dan Coombs <dan.coombs@alchemy.com>
Date:   Fri Feb 9 14:56:12 2024 -0500

    feat(builder): only require bundle fees when replacing (#590)

commit 0b2aa762432d35236839df6d6a50a7b26cec16a7
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Tue Feb 6 13:14:53 2024 -0500

    feat(tracer): add RIP-7212 precompile (#586)

commit bdf15108afafbc4f44c7b774a47b276b20e682e8
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Tue Feb 6 09:33:38 2024 -0500

    feat(chains): add polygon amoy and use alloy-rs (#583)

commit a670f081593d8362bf74f94bef36cf0ca9b3513c
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Fri Feb 2 16:27:14 2024 -0500

    feat(paymaster): add paymaster withdrawal logic to tracker (#572)

commit 96bd95bf065eec05e9e8ef636e699dbd39b80cdb
Author: JP <36560907+0xfourzerofour@users.noreply.github.com>
Date:   Fri Feb 2 13:20:27 2024 -0500

    feat(admin): create admin namespace and add paymaster clear / toggle for reputation (#570)

    Co-authored-by: Alex Miao <alex.miao@alchemy.com>

mmontour1306 committed:
Merge cleanup
 Changes to be committed:
	deleted:    .github/workflows/docker-images.yml
	deleted:    crates/provider/src/ethers/nonce_manager.rs
	deleted:    crates/provider/src/traits/nonce_manager.rs

mmontour1306 committed:
Fix contract building, adding @gnosis.pm remapping to https://github.com/safe-global/safe-smart-account.git

 Changes to be committed:
	modified:   .gitmodules
	modified:   crates/types/build.rs
	new file:   crates/types/contracts/lib/safe-smart-account

mmontour1306 committed:
Merge branch 'boba-develop' into merge-v030

mmontour1306 committed:
Merge branch 'boba-develop' into merge-v030

mmontour1306 committed:
Change "docker-compose" to "docker compose"

 Changes to be committed:
	modified:   test/spec-tests/launchers/rundler-launcher/rundler-launcher.sh
	modified:   test/spec-tests/local/launcher.sh
	modified:   test/spec-tests/remote/launcher.sh

mmontour1306 committed:
Merge branch 'boba-develop' into merge-v030

mmontour1306 committed:
Set toolchain version
 Changes to be committed:
	modified:   .github/workflows/release.yaml
	modified:   .github/workflows/unit.yaml

mmontour1306 committed:
Cleanup - "cargo fmt"
 Changes to be committed:
	modified:   bin/rundler/src/cli/mod.rs
	modified:   crates/builder/src/bundle_proposer.rs
	modified:   crates/builder/src/bundle_sender.rs
	modified:   crates/builder/src/transaction_tracker.rs
	modified:   crates/provider/src/ethers/entry_point/v0_6.rs
	modified:   crates/provider/src/ethers/entry_point/v0_7.rs
	modified:   crates/provider/src/ethers/provider.rs
	modified:   crates/provider/src/traits/entry_point.rs
	modified:   crates/provider/src/traits/test_utils.rs
	modified:   crates/rpc/src/eth/api.rs
	modified:   crates/rpc/src/eth/router.rs
	modified:   crates/rpc/src/task.rs
	modified:   crates/sim/src/estimation/estimate_verification_gas.rs
	modified:   crates/sim/src/estimation/v0_6.rs
	modified:   crates/sim/src/gas/gas.rs
	modified:   crates/sim/src/simulation/v0_6/tracer.rs
	modified:   crates/types/src/hybrid_compute.rs
	modified:   crates/types/src/user_operation/v0_6.rs
	modified:   crates/types/src/user_operation/v0_7.rs

mmontour1306 committed:
More "cargo fmt" fixes.
 Changes to be committed:
	modified:   bin/rundler/src/cli/mod.rs
	modified:   crates/builder/src/bundle_proposer.rs
	modified:   crates/builder/src/bundle_sender.rs
	modified:   crates/builder/src/task.rs
	modified:   crates/provider/src/ethers/entry_point/v0_6.rs
	modified:   crates/rpc/src/eth/api.rs
	modified:   crates/sim/src/estimation/mod.rs
	modified:   crates/sim/src/simulation/v0_6/tracer.rs
	modified:   crates/types/src/hybrid_compute.rs

mmontour1306 committed:
Some "cargo clippy" fixes.
 Changes to be committed:
	modified:   crates/builder/src/bundle_proposer.rs
	modified:   crates/rpc/src/eth/api.rs
	modified:   crates/rpc/src/eth/router.rs
	modified:   crates/types/src/hybrid_compute.rs

mmontour1306 committed:
More "clippy" fixes.
 Changes to be committed:
	modified:   crates/types/src/hybrid_compute.rs

mmontour1306 committed:
fix: remove actions-rs to resolve deprecation warnings.

 Changes to be committed:
	modified:   .github/workflows/ci.yaml

mmontour1306 committed:
fix: point to a bobanetwork bundler-test-executor repo
 Changes to be committed:
	modified:   .github/workflows/compliance.yaml

wsdt commented:
Did the Docker image work for you @mmontour1306 ? Not able to get it running. 

Did the startup command possibly change? How did you get it up?

mmontour1306 committed:
fix: Docker containers were broken.

 Changes to be committed:
	modified:   hybrid-compute/Dockerfile.offchain-rpc
	modified:   hybrid-compute/docker-compose.yml

wsdt commented:
@mmontour1306 after the merge, when do we plan to release a new version to DockerHub (for e.g. the aa-hc-example)?

wsdt commented:
Container and example still work for me!

mmontour1306 committed:
fix: reverted an unneeded change in the AA Account contracts
 Changes to be committed:
	modified:   crates/types/contracts/lib/account-abstraction-versions/v0_6
